packages/astro/src/core/config/schemas/refined.ts

```diff
@@ -203,4 +203,35 @@ export const AstroConfigRefinedSchema = z.custom<AstroConfig>().superRefine((con
 			}
 		}
 	}
+
+	if (config.experimental.csp && typeof config.experimental.csp === 'object') {
+		const { scriptHashes, styleHashes } = config.experimental.csp;
+		if (scriptHashes) {
+			for (const hash of scriptHashes) {
+				if (
+					!(hash.startsWith('sha256-') || hash.startsWith('sha384-') || hash.startsWith('sha512-'))
```

Comment:
Instead of harcoding, I think it could be beneficial to have a mapping object in `csp/config.ts` eg:
```ts
const ALGORITHMS = {
	"SHA-256": "sha256"
	// ...
} as const
```
And then use those values for the algorithm schema, types etc

Reply:
Is there another place where this gets convenient? I don't want to make things generic if they are used in one single place